test/unit/specs/hive/list.js

```diff
@@ -0,0 +1,533 @@
+import { expect } from 'chai'
+import { describe, it } from 'mocha'
+
+import Backendless, { APP_PATH, forTest, prepareMockRequest } from '../../helpers/sandbox'
+import { HiveTypes } from '../../../../src/hive/constants'
+
+describe('List Store', function() {
+
+  forTest(this)
+
+  const hiveName = 'test'
+
+  describe('Methods', () => {
+    const fakeResult = { foo: true }
+
+    const storeKey = 'testStoreKey'
+    const baseUrl = `${APP_PATH}/hive/${hiveName}/${HiveTypes.LIST}/${storeKey}`
+
+    let store
+
+    beforeEach(() => {
+      store = Backendless.Hive(hiveName).ListStore(storeKey)
+    })
+
+    describe('Get', () => {
+      it('success', async () => {
+        const req1 = prepareMockRequest(fakeResult)
+
+        const result1 = await store.get()
+
+        expect(req1).to.deep.include({
+          method: 'GET',
+          path  : `${baseUrl}`,
+        })
+
+        expect(result1).to.be.eql(fakeResult)
+      })
+
+      it('success with index', async () => {
+        const req1 = prepareMockRequest(fakeResult)
+
+        const result1 = await store.get(0)
+
+        expect(req1).to.deep.include({
+          method: 'GET',
+          path  : `${baseUrl}/0`,
+        })
+
+        expect(result1).to.be.eql(fakeResult)
+      })
+
+      it('success with range', async () => {
+        const req1 = prepareMockRequest(fakeResult)
+
+        const result1 = await store.get(0, 3)
+
+        expect(req1).to.deep.include({
+          method: 'GET',
+          path  : `${baseUrl}?from=0&to=3`,
+        })
+
+        expect(result1).to.be.eql(fakeResult)
+      })
+
+      it('fails when store created without key', async () => {
+        store = Backendless.Hive(hiveName).ListStore()
+
+        const errorMsg = 'Store must be created with store key.'
+
+        await expect(() => store.get()).to.throw(errorMsg)
+      })
+
+      it('fails when arguments is invalid', async () => {
```

Comment:
I would split it on 3 tests

Reply:
done, also made same with other similar tests